frontend/integration-tests/views/devconsole-view/template.view.ts

```diff
@@ -0,0 +1,64 @@
+// import * as _ from 'lodash';
+import { $, $$, browser, by, ExpectedConditions as until } from 'protractor';
+
+import { appHost, testName, waitForNone } from '../../protractor.conf';
+import { resourceTitle } from '../../views/crud.view';
+
+
+export const reloadBtn = $('#reload-object');
+
+//element finder with class name
+export const untilLoadingBoxLoaded = until.presenceOf($('.loading-box__loaded'));
+export const untilNoLoadersPresent = waitForNone($$('.co-m-loader'));
+
+export const isLoaded = () => browser.wait(until.and(untilNoLoadersPresent, untilLoadingBoxLoaded)).then(() => browser.sleep(1000));
+
+export const resourceRows = $$('[data-test-rows="resource-row"]');
+// export const resourceRowNamesAndNs = $$('.co-m-resource-icon + a');
+
+export const rowForName = (name: string) => resourceRows.filter((row) => row.$$('.co-m-resource-icon + a').first().getText().then(text => text === name)).first();
+
+//element finder with data-test-id name
+const actionsMenu = $('[data-test-id="details-actions"]');
```

Comment:
Question - Josh Wilson showed me the use of "data-test-id" last week - does this replace all use of "id="?

Reply:
If the `data-test-id` attribute is present then you can use it instead of `id`, but you can't assume it will be there. And it doesn't work like a substitute.